jax/_src/core.py

```diff
@@ -2047,6 +2047,70 @@ def dimension_as_value(d: DimSize):
   if hasattr(d, "dimension_as_value"): return d.dimension_as_value()
   return operator.index(d)
 
+def canonicalize_slice(
```

Comment:
Is it worth testing this in isolation?

Reply:
There are really very extensive tests for this, both as part of regular JAX, and in the context of shape polymorphism.  Those are not quite unit tests, but I think they are still small enough.